README.md

```diff
@@ -0,0 +1,187 @@
+To build Ruby image, choose between CentOS or RHEL based image:
+*  **RHEL based image**
+    This image is not available as trusted build in [Docker Index](https://index.docker.io).
+
+    To build a rhel-based ruby-2.0 image, you need to run the build on properly
+    subscribed RHEL machine.
+
+    ```
+    $ git clone https://github.com/openshift/sti-ruby.git
+    $ cd sti-ruby
+    $ make build TARGET=rhel7 VERSION=2.0
+    ```
+
+*  **CentOS based image**
+    ```
+    $ git clone https://github.com/openshift/sti-ruby.git
+    $ cd sti-ruby
+    $ make build VERSION=2.0
+    ```
+
+**Notice: By omitting the `VERSION` parameter, the build/test action will be performed
+on all the supported versions of Ruby. Since we are now supporting only version `2.0`,
+you can omit this parameter.**
+
+
+Usage
+---------------------
+To build a simple [sinatra-app](https://github.com/jhadvig/sinatra-app-example) application,
+using standalone [STI](https://github.com/openshift/source-to-image) and then run the
+resulting image with [Docker](http://docker.io) execute:
+
+*  **For RHEL based image**
+    ```
+    $ sti build https://github.com/jhadvig/sinatra-app-example.git openshift/ruby-20-rhel7 sinatra-app
+    $ docker run -p 9292:9292 sinatra-app
+    ```
+
+*  **For CentOS based image**
+    ```
+    $ sti build https://github.com/jhadvig/sinatra-app-example.git openshift/ruby-20-centos7 sinatra-app
+    $ docker run -p 9292:9292 sinatra-app
+    ```
+
+**Accessing the application:**
+```
+$ curl 127.0.0.1:9292
+```
+
+
+Test
+---------------------
+This repository also provides [STI](https://github.com/openshift/source-to-image) test framework,
+which launches tests to check functionality of a simple ruby application built on top of sti-ruby image.
+
+User can choose between testing ruby test application based on RHEL or CentOS image.
+
+*  **RHEL based image**
+
+    This image is not available as trusted build in [Docker Index](https://index.docker.io).
+
+    To test a rhel7-based ruby-2.0 image, you need to run the test on a properly
+    subscribed RHEL machine.
+
+    ```
+    $ cd sti-ruby
+    $ make test TARGET=rhel7 VERSION=2.0
+    ```
+
+*  **CentOS based image**
+
+    ```
+    $ cd sti-ruby
+    $ make test VERSION=2.0
+    ```
+
+**Notice: By omitting the `VERSION` parameter, the build/test action will be performed
+on all the supported versions of Ruby. Since we are now supporting only version `2.0`
+you can omit this parameter.**
+
+
+Repository organization
+------------------------
+* **`<ruby-version>`**
+
+    * **Dockerfile**
+
+        CentOS based Dockerfile.
+
+    * **Dockerfile.rhel7**
+
+        RHEL based Dockerfile. In order to perform build or test actions on this
+        Dockerfile you need to run the action on properly subscribed RHEL machine.
+
+    * **`.sti/bin/`**
+
+        This folder contains scripts that are run by [STI](https://github.com/openshift/source-to-image):
+
+        *   **assemble**
```

Comment:
somewhere in here you should discuss the ENV variables/marker files that affect behavior of RUN and ASSEMBLE